common/model-views.cpp

```diff
@@ -2166,89 +2205,267 @@ namespace rs2
                 }
-                else
+
+                ImGui::SameLine();
+
+                label = to_string() << "FPS: " << std::setprecision(2) << std::setw(7) << std::fixed << fps.get_fps();
+                ImGui::Text("%s", label.c_str());
+                if (ImGui::IsItemHovered())
                 {
-                    ImGui::SetTooltip("Timestamp: Hardware Clock");
+                    ImGui::SetTooltip("%s", "FPS is calculated based on timestamps and not viewer time");
                 }
+
+                line_y += ImGui::GetTextLineHeight() + 5;
             }
+        }
 
-            ImGui::SameLine();
+        
 
-            label = to_string() << " Frame: " << std::left <<  frame_number;
-            ImGui::Text("%s", label.c_str());
+        if (show_metadata)
+        {
+            std::vector<attribute> stream_details;
+
+            if (true) // Always show stream details options
+            {
+                stream_details.push_back({ "Frame Timestamp",
+                    to_string() << std::fixed << std::setprecision(1) << timestamp, 
+                    "Frame Timestamp is normalized represetation of when the frame was taken.\n"
+                    "It's a property of every frame, so when exact creation time is not provided by the hardware, an approximation will be used.\n"
+                    "Clock Domain feilds helps interpret the meaning of timestamp\n"
+                    "Timestamp is measured in milliseconds, and is allowed to roll-over (reset to zero) in some situations" });
+                stream_details.push_back({ "Clock Domain",
+                    to_string() << rs2_timestamp_domain_to_string(timestamp_domain), 
+                    "Clock Domain describes the format of Timestamp feild. It can be one of the following:\n"
+                    "1. System Time - When no hardware timestamp is available, system time of arrival will be used.\n"
+                    "                 System time benefits from being comparable between device, but suffers from not being able to approximate latency.\n"
+                    "2. Hardware Clock - Hardware timestamp is attached to the frame by the device, and is consistent accross device sensors.\n"
+                    "                    Hardware timestamp encodes percisely when frame was captured, but cannot be compared across devices\n"
+                    "3. Global Time - Global time is provided when the device can both offer hardware timestamp and implements Global Timestamp Protocol.\n"
+                    "                 Global timestamps encode exact time of capture and at the same time are comparable accross devices." });
+                stream_details.push_back({ "Frame Number",
+                    to_string() << frame_number, "Frame Number is a rolling ID assigned to frames.\n"
+                    "Most device do not guaranty consequitive frames to have conseuquitive frame numbers\n"
```

Comment:
Most device**s** do not guarant**ee** consecutive frames to have sequential frame numbers
devices... guarantee